### ISSUE TITLE
NullPointerException when any benchmark is not matched by :selector predicate

### ISSUE THREAD

**Issue by glts** [jgpc42/jmh-clojure#1]
The selector functionality activated via the `:select` option seems to
be broken: a `NullPointerException` is thrown whenever the active
selector’s predicate returns false for a benchmark.

```
NullPointerException
        jmh.exec/run* (exec.clj:95)
        jmh.exec/run* (exec.clj:74)
        jmh.exec/run (exec.clj:208)
        jmh.exec/run (exec.clj:181)
        clojure.core/comp/fn--4727 (core.clj:2460)
        clojure.core/comp/fn--4727 (core.clj:2460)
        clojure.core/comp/fn--4727 (core.clj:2461)
        jmh.core/run (core.clj:87)
        jmh.core/run (core.clj:14)
        jmh-example2.core/eval10215 (form-init3157860254002531894.clj:1)
        jmh-example2.core/eval10215 (form-init3157860254002531894.clj:1)
        clojure.lang.Compiler.eval (Compiler.java:6927)
```

To reproduce, run a benchmark with the option `:select :example` and the
following selector specification.

```
:selectors
{:example (constantly false)}
```

**Comment by jgpc42** [comment]
The NPE was caused by jmh's `NoBenchmarksException` returning `nil` from `getMessage`, which I didn't expect. I have fixed the issue in my local repo.

Do you have an opinion on how the larger "no benchmarks" issue should be handled? Should the `NoBenchmarksException` be re-thrown? Or should `jmh.core/run` just return an empty seq in this case?

Thanks again for the bug report and good repro!

**Comment by jgpc42** [comment]
I've thought a bit on this now and I'm leaning on just re-throwing the `NoBenchmarksException` for the time being,  which is consistent with the JMH API. Also, `jmh.exec/run` already will throw other JMH exception types. For example, a `RunnerException` is thrown when JMH can't acquire the lock file.

Returning an empty seq might be confusing too if a user simply misspelled a selector name for example.

**Comment by glts** [comment]
Thank you for answer and work.

Unfortunately I don’t think I can answer these questions. I am only a
user; I’m not familiar with the JMH APIs. This issue here I stumbled
over when I wanted to run a single benchmark and did not manage to write
a selector that worked.

**Comment by jgpc42** [comment]
Fixed with 194a5ac. Released version 0.1.2. Thanks again!
